jmetal-core/src/main/java/org/uma/jmetal/solution/impl/DefaultDoubleSolution.java

```diff
@@ -77,4 +74,11 @@ public DefaultDoubleSolution copy() {
   public String getVariableValueString(int index) {
     return getVariableValue(index).toString() ;
   }
+  
+  private void initializeDoubleVariables() {
+    for (int i = 0 ; i < problem.getNumberOfVariables(); i++) {
+      Double value = randomGenerator.nextDouble(getLowerBound(i), getUpperBound(i)) ;
```

Comment:
Here you remove the lower bound, which is wrong unless this lower bound is zero. If there is a lower and upper bound, then the value should fall in between, which is achieved by computing "min + rand x (max - min)", so if rand = 0 the result is min, if it is 1 the result is max, and otherwise it is in between.